src/resources/classes/asset.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 Cognite AS
+
+import { CogniteClient } from '../..';
+import * as types from '../../types/types';
+import { AssetList } from './assetList';
+
+export interface SubtreeOptions {
+  depth?: number;
+}
+export interface DeleteOptions {
+  recursive?: boolean;
+}
+export class Asset implements types.Asset {
+  public id: types.CogniteInternalId;
+  public externalId?: types.CogniteExternalId;
+  public parentId?: types.CogniteInternalId;
+  public name: types.AssetName;
+  public description?: types.AssetDescription;
+  public metadata?: types.Metadata;
+  public source?: types.AssetSource;
+  public lastUpdatedTime: Date;
+  public createdTime: Date;
+  public rootId: types.CogniteInternalId;
+  private client: CogniteClient;
+
+  constructor(client: CogniteClient, props: types.Asset) {
+    this.client = client;
+    this.id = props.id;
+    this.externalId = props.externalId;
+    this.parentId = props.parentId;
+    this.name = props.name;
+    this.description = props.description;
+    this.metadata = props.metadata;
+    this.source = props.source;
+    this.lastUpdatedTime = props.lastUpdatedTime;
+    this.createdTime = props.createdTime;
+    this.rootId = props.rootId;
+  }
+
+  public delete = (options: DeleteOptions = {}) => {
+    return this.client.assets.delete(
+      [
+        {
+          id: this.id,
+        },
+      ],
+      options
+    );
+  };
+
+  public parent: () => Promise<null | Asset> = async () => {
+    const { parentId } = this;
```

Comment:
destructive assignment for only one property?
maybe just `if (this.parentId) {...} else { return null }`